src/wh_server_she.c

```diff
@@ -61,34 +57,65 @@ enum WOLFHSM_SHE_SB_STATE {
     WOLFHSM_SHE_SB_SUCCESS,
     WOLFHSM_SHE_SB_FAILURE,
 };
-uint8_t hsmShePrngState[WOLFHSM_SHE_KEY_SZ];
-uint8_t hsmSheSbState = WOLFHSM_SHE_SB_INIT;
-uint8_t hsmSheCmacKeyFound = 0;
-uint8_t hsmSheRamKeyPlain = 0;
-uint32_t hsmSheBlSize = 0;
-uint32_t hsmSheBlSizeReceived = 0;
-uint32_t hsmSheInitRng = 0;
+static uint8_t hsmShePrngState[WOLFHSM_SHE_KEY_SZ];
+static uint8_t hsmSheSbState = WOLFHSM_SHE_SB_INIT;
+static uint8_t hsmSheCmacKeyFound = 0;
+static uint8_t hsmSheRamKeyPlain = 0;
+static uint8_t hsmSheUidSet = 0;
+static uint32_t hsmSheBlSize = 0;
+static uint32_t hsmSheBlSizeReceived = 0;
+static uint32_t hsmSheRndInited = 0;
 /* cmac is global since the bootloader update can be called multiple times */
 Cmac sheCmac[1];
+Aes sheAes[1];
+
+static int isLittleEndian() {
+    unsigned int x = 1; /* 0x00000001 */
+    char *c = (char*)&x;
+    return (int)*c;
+}
+
+/* Converts a 32-bit value from host to network byte order */
+static uint32_t htonl(uint32_t hostlong) {
+    if (isLittleEndian()) {
+        return ((hostlong & 0x000000FF) << 24) | ((hostlong & 0x0000FF00) << 8)
+            | ((hostlong & 0x00FF0000) >> 8) | ((hostlong & 0xFF000000) >> 24);
```

Comment:
Consider using the equally ugly functions in wh_comm.h.  We need to move to a utility library at some point.  low priority,